primitives/core/src/traits.rs

```diff
@@ -174,6 +174,19 @@ pub trait BareCryptoStore: Send + Sync {
 	}
 }
 
+/// VRF-signing capability.
+pub trait VRFSigner {
```

Comment:
Move this to `BareCryptoStore` or add `VRFSignerPtr` that is accepted by the functions above.

Reply:
Rust generally does not use `Ptr` in type names.

Reply:
In this file, we have `BareCryptoStorePtr` which is `Arc<RwLock<dyn BareCryptoStore>>` so the idea was to have a similar pointer for `VRFSigner`... the idea i am trying to convey is whether it makes sense to split `vrf_sign` method from `BareCryptoStore` or not.